### PR TITLE
feat (query): support for absent_over_time for subqueries

### PR DESCRIPTION
**Pull request by kvpetrov**
**Pull Request checklist**

- [ ] The commit(s) message(s) follows the contribution [guidelines](CONTRIBUTING.md) ?
- [ ] Tests for the changes have been added (for bug fixes / features) ?
- [ ] Docs have been added / updated (for bug fixes / features) ?

**Current behavior :** (link exiting issues here : https://help.github.com/articles/basic-writing-and-formatting-syntax/#referencing-issues-and-pull-requests)

**New behavior :**

**BREAKING CHANGES**

If this PR contains a breaking change, please describe the impact and migration
path for existing applications.
If not please remove this section.

Breaking changes may include:
- Any schema changes to any Cassandra tables
- The serialized format for Dataset and Column (see .toString methods)
- Over the wire formats for Akka messages / case classes
- Changes to the HTTP public API
- Changes to query parsing / PromQL parsing

**Other information**:

### CODE REVIEW

File: query/src/main/scala/filodb/query/exec/rangefn/AggrOverTimeFunctions.scala [hunk 1]
```diff
@@ -405,7 +405,11 @@ class AvgOverTimeFunction(var sum: Double = Double.NaN, var count: Int = 0) exte
   override def apply(startTimestamp: Long, endTimestamp: Long, window: Window,
                      sampleToEmit: TransientRow,
                      queryConfig: QueryConfig): Unit = {
-    sampleToEmit.setValues(endTimestamp, sum/count)
+    if (count == 0) {
+      sampleToEmit.setValues(endTimestamp, Double.NaN)
+    } else {
+      sampleToEmit.setValues(endTimestamp, sum/count)
+    }
```
Comment (vishramachandran) [inline query/src/main/scala/filodb/query/exec/rangefn/AggrOverTimeFunctions.scala:412]: This should be reverted. There is no divide-by-zero exception when Double is involved. Result is Infinity. 

```
scala> 5d/0d
res0: Double = Infinity
```

File: coordinator/src/main/scala/filodb.coordinator/queryplanner/SingleClusterPlanner.scala [hunk 2]
```diff
@@ -257,28 +257,74 @@ class SingleClusterPlanner(dsRef: DatasetRef,
...
-        paramsExec,
-        sqww.offsetMs,
-        false,
-        true
+        Seq(),
+        offsetMs,
+        false
+      ))
+    )
+    val aggregate = Aggregate(AggregationOperator.Sum, innerPlan, Nil, Seq("job"))
```
Comment (vishramachandran) [inline coordinator/src/main/scala/filodb.coordinator/queryplanner/SingleClusterPlanner.scala:308]: Can you remove code duplication? I think we may have the same thing for regular absent-over-time function as well.

**Review by vishramachandran — APPROVED** [review]
Approved with comments.
Please wait for @TanviBhavsar to review as well. She is aware of Absent intricacies better than me.

File: query/src/main/scala/filodb/query/exec/rangefn/AggrOverTimeFunctions.scala (hunk 1 above)
Comment (amolnayak311) [reply query/src/main/scala/filodb/query/exec/rangefn/AggrOverTimeFunctions.scala:412]: Hi @vishramachandran What would be the expected behavior? If sum is NaN the division will give NaN irrespective of whether count is 0 or not, but will be Infinity when sum is non NaN and count is 0. The code change above will give NaN in all scenarios.

Comment (kvpetrov) [reply query/src/main/scala/filodb/query/exec/rangefn/AggrOverTimeFunctions.scala:412]: the problem is that it does not match with the Prometheus, the reason, I had to put this in is to match with the Prom behavior. I don't think conceptually -infinity is correct either.

**Review by TanviBhavsar — APPROVED** [review]
Looks good to me. Can you also add a unit test?

**Pull request merged**